src/server/publicApi.js

```diff
@@ -60,14 +56,37 @@ const runPublicApi = (gameServer) => {
     };
   });
 
-  router.get('/model/:id', async (ctx) => {
-    const matchID = ctx.params.id;
+  //authorise
+  router.use('/:matchID/', async (ctx, next) => {
+    const user = auth(ctx);
+    const game = await gameServer.db.fetch(ctx.params.matchID, { metadata: true });
+    const metadata = game.metadata;
+    if(user && metadata && metadata.players && user.pass === metadata.players[user.name].credentials) {
+      return await next();
```

Comment:
I think it is a great solution writing this middleware!

Reply:
Thanks!